runtime/lua/vim/_defaults.lua

```diff
@@ -155,6 +155,30 @@ do
     end,
   })
 
+  vim.api.nvim_create_autocmd('TermRequest', {
+    group = nvim_terminal_augroup,
+    desc = 'Check for an OSC foreground/background color request, and respond accordingly',
+    callback = function(args)
+      local fg_request = args.data == '\x1b]10;?'
+      local bg_request = args.data == '\x1b]11;?'
+      if fg_request or bg_request then
+        -- WARN: This does not return the actual foreground/background color,
+        -- but rather returns:
+        --   - fg=white/bg=black when Nvim option 'background' is 'dark'
+        --   - fg=black/bg=white when Nvim option 'background' is 'light'
+        local channel = vim.api.nvim_get_option_value('channel', { buf = args.buf })
```

Comment:
This currently assumes that the current buffer is the terminal buffer that initiated the `TermRequest`, which isn't always the case.

I switched this PR to a draft about 20 minutes ago thinking that there would be some update necessary. Shortly after I switched away from a draft and opened Issue #26972.

Is there currently a way to get the terminal buffer number that initiated the `TermRequest`?

Related: @gpanders's comments here: https://github.com/neovim/neovim/pull/22159#discussion_r1444018763

Reply:
For now, I've limited the OSC response handling to run only when the terminal buffer is the current buffer.

Reply:
See https://github.com/neovim/neovim/pull/26974

Reply:
> "For now, I've limited the OSC response handling to run only when the terminal buffer is the current buffer."

This handling is now reverted.